untrusted/test_enclave/build.rs

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) 2022 The MobileCoin Foundation
```

Comment:
We already have code that does everything in this build script in the main repo. If we need to refactor it out into this repo (or another repo), we should just do that.

Reply:
My initial thought was that this untrusted crate(s) would eventually be utilized to help test the trusted crate(s) behavior as well as any trusted code. 
I was in the mindset that actually building and signing of an enclave would reside in the trusted crate(s) and so to prevent a circular dependency the enclave building logic was crudely repeated here for testing use.

Perhaps,  there should be 3 groups of crates:
* untrusted
* trusted
* sgx_types and enclave building logic

Reply:
My thinking here is that we're going to end up needing a collection of 4 crates for each `libfoo` provided by the SDK:

1. FFI types (bindgen)
2. FFI methods (bindgen)
3. Rust types (wrapping FFI types)
4. Rust APIs (wrapping FFI methods)

Like, there's zero need to link against an EPID crate in a DCAP world, and we're not doing any switchless stuff in mobilecoin.git (so we shouldn't have those functions in the enclave).

But otherwise, you're right, there is a "last" type for all the build utilities.

Reply:
This is now broken up into 3 crates:
* mc-sgx-urts-sys-types
* mc-sgx-urts-sys
* mc-sgx-urts

I went with `sys` for the name to match, from https://doc.rust-lang.org/cargo/reference/build-scripts.html#-sys-packages
For urts I'm not fully sure what types would be independent from the API.